File: typescript/src/token.ts
export const TokenTypes = {
    ILLEGAL: "ILLEGAL",
    EOF: "EOF",
    // Identifiers + literals
    IDENT: "IDENT",
    INT: "INT",
    // Operators
    ASSIGN: "=",
    PLUS: "+",
	MINUS: "-",
	ASTERISK: "*",
	SLASH: "/",
	BANG: "!",
	LT: "<",
	GT: ">",
    EQ: "==",
    NOT_EQ: "!=",
    // Delimiters
    COMMA: ",",
    SEMICOLON: ";",
    LPAREN: "(",
    RPAREN: ")",
    LBRACE: "{",
    RBRACE: "}",
    // Keywords
    FUNCTION: "FUNCTION",
    LET: "LET",
    IF: "IF",
    ELSE: "ELSE",
    RETURN: "RETURN",
    TRUE: "TRUE",
    FALSE: "FALSE",
} as const;

export type TokenType = typeof TokenTypes[keyof typeof TokenTypes];

export type Token = {
    type: TokenType,
    literal: string
}

export function createToken(type: TokenType, literal: string): Token {
    return { type, literal };
}

File: java/src/main/java/interpreter/Token.java
package interpreter;

public enum Token {
    ILLEGAL,
    EOF("eof"),
    // Identifiers + literals
    IDENT,
    INT(""),
    // Operators
    ASSIGN("="),
    EQ("=="),
    NOT_EQ("!="),
    LT("<"),
    GT(">"),
    BANG("!"),
    PLUS("+"),
    MINUS("-"),
    SLASH("/"),
    ASTERISK("*"),
    // Delimiters
    COMMA(","),
    SEMICOLON(";"),
    LPAREN("("),
    RPAREN(")"),
    LBRACE("{"),
    RBRACE("}"),
    // Keywords
    FUNCTION("fn"),
    LET("let"),
    IF("if"),
    ELSE("else"),
    TRUE("true"),
    FALSE("false"),
    RETURN("return");

    private String literal;

    Token(String literal) {
        this.literal = literal;
    }

    Token() {
        this.literal = "";
    }

    Token literal(String literal) {
        this.literal = literal;
        return this;
    }

    String getLiteral() {
        return this.literal;
    }
}
